ttl/ttlworker/job_manager.go

```diff
@@ -749,64 +608,38 @@ func (m *JobManager) lockNewJob(ctx context.Context, se session.Session, table *
 	if err != nil {
 		return nil, err
 	}
-	return m.createNewJob(expireTime, now, table)
+	return m.createNewJob(now, table)
 }
 
-func (m *JobManager) createNewJob(expireTime time.Time, now time.Time, table *cache.PhysicalTable) (*ttlJob, error) {
+func (m *JobManager) createNewJob(now time.Time, table *cache.PhysicalTable) (*ttlJob, error) {
 	id := m.tableStatusCache.Tables[table.ID].CurrentJobID
 
-	statistics := &ttlStatistics{}
-
-	ranges, err := table.SplitScanRanges(m.ctx, m.store, splitScanCount)
-	if err != nil {
-		return nil, err
-	}
-
-	jobCtx, cancel := context.WithCancel(m.ctx)
-
-	scanTasks := make([]*ttlScanTask, 0, len(ranges))
-	for _, r := range ranges {
-		scanTasks = append(scanTasks, &ttlScanTask{
-			ctx:        jobCtx,
-			tbl:        table,
-			expire:     expireTime,
-			scanRange:  r,
-			statistics: statistics,
-		})
-	}
-
 	return &ttlJob{
 		id:      id,
 		ownerID: m.id,
 
-		ctx:    jobCtx,
-		cancel: cancel,
-
 		createTime: now,
 		// at least, the info schema cache and table status cache are consistent in table id, so it's safe to get table
 		// information from schema cache directly
-		tbl:   table,
-		tasks: scanTasks,
+		tbl: table,
 
-		status:     cache.JobStatusWaiting,
-		statistics: statistics,
+		status: cache.JobStatusWaiting,
 	}, nil
 }
 
 // updateHeartBeat updates the heartbeat for all task with current instance as owner
-func (m *JobManager) updateHeartBeat(ctx context.Context, se session.Session) error {
-	now := se.Now()
+func (m *JobManager) updateHeartBeat(ctx context.Context, se session.Session, now time.Time) error {
 	for _, job := range m.localJobs() {
+		if job.createTime.Add(ttlJobTimeout).Before(now) {
```

Comment:
Why just stop update hb here and some job termination codes are removed when job is time out?

Reply:
It's a mistake. They are terminated now.